rmf_fleet_adapter_python/scripts/rmf_fleet_adapter_python/test_adapter.py

```diff
@@ -75,150 +104,177 @@ def __init__(self, name):
         )
 
     def reset(self, name=None):
+        # reset sets a new name if provided
         if name is not None:
             self.name = name
 
-        self.tasks = {}
-        self.success_flag = False
-
-        try:
-            self.timer.reset()
-            self.timer.cancel()
-        except Exception:
-            pass
-
-        self.timer = None
+        self.tasks = []
 
     def _process_request_cb(self, msg):
-        # Process request if addressed to this dispenser
+        # Process request only if addressed to this dispenser
         if msg.target_guid != self.name:
             return
 
-        print("[MockQuietDispenser] REQUEST RECEIVED")
-        if self.tasks.get(msg.request_guid):
-            print("QUIET DISPENSER SUCCESS")
-            status = DISPENSER_RESULT_SUCCESS
-        else:
-            self.tasks[msg.request_guid] = False
-            status = DISPENSER_RESULT_ACKNOWLEDGED
-
-            self.timer = self.create_timer(
-                0.01,
-                partial(self._timer_cb, msg=msg)
-            )
-
-        result = dispenser_msgs.DispenserResult()
-        result.time = self.get_clock().now().to_msg()
-        result.status = status
-        result.source_guid = self.name
-        result.request_guid = msg.request_guid
+        task_already_received = [
+            x for x in self.tasks if x.request_guid == msg.request_guid]
 
-        self.result_pub.publish(result)
+        if task_already_received:
+            # print("[MockDispenser] DUPLICATE REQUEST")
+            pass
+        else:
+            # New request to process
+            print("[MockDispenser] NEW REQUEST RECEIVED")
+            new_task = DispenserTask(
+                request_guid=msg.request_guid,
+                receive_time=self.get_clock().now().to_msg())
+            self.tasks.append(new_task)
 
-    def _timer_cb(self, msg):
-        if not self.timer:
+    def _timer_cb(self):
+        # Report Dispenser State
+        if self.publish_states:
+            state = dispenser_msgs.DispenserState()
+            state.time = self.get_clock().now().to_msg()
+            state.guid = self.name
+            if self.tasks:
+                state.request_guid_queue = [
+                    x.request_guid for x in self.tasks][
+                        self.current_request_idx:]
+            if state.request_guid_queue:
+                state.mode = DISPENSER_STATE_BUSY
+            else:
+                state.mode = DISPENSER_STATE_IDLE
```

Comment:
e.g. `state_msg.BUSY`, and `state_msg.IDLE`

Reply:
done!